tests/pfcwd/test_pfcwd_all_port_storm.py

```diff
@@ -113,6 +114,108 @@ def set_storm_params(duthost, fanout_graph, fanouthosts, peer_params):
+    src_dut_intf = random.choice(list(non_vlan_ports))
+    dest_mac = dut.get_dut_iface_mac(src_dut_intf)
+    # Find out if the selected port is a lag member
+    # If so, we need to use the neighbor address of the portchannel.
+    # else, we need the neighbor address of the interface itself.
+    required_intf = dst_dut_intf
+    for intf in mg_facts['minigraph_portchannels']:
+        if dst_dut_intf in mg_facts['minigraph_portchannels'][intf]['members']:
+            required_intf = intf
+            break
+    # At this point, required_intf is either a portchannel or Ethernet port.
+    # It should have a neibhor address or it is an error.
+    dst_ip_addr = None
+    for intf_obj in all_ip_intfs:
+        if intf_obj['attachto'] == required_intf:
+            dst_ip_addr = intf_obj['peer_addr']
+            break
+    if dst_ip_addr is None:
+        raise RuntimeError("Couldnot find the neighbor address for intf:{}".format(required_intf))
+    ptf_src_port = mg_facts['minigraph_ptf_indices'][src_dut_intf]
+    ptf_dst_port = mg_facts['minigraph_ptf_indices'][dst_dut_intf]
+    extra_vars = {
+        f'{program_name}_args':
+            'dest_mac=u"{}";dst_ip_addr={};ptf_src_port={};ptf_dst_port={};pfc_queue_idx={}'.format(
+                dest_mac,
+                dst_ip_addr,
+                ptf_src_port,
+                ptf_dst_port,
+                3   # Hardcoded in the testcase as well.
+                )}
+    try:
+        ptfhost.command('supervisorctl stop {}'.format(program_name))
+    except BaseException:
+        pass
+
+    ptfhost.host.options["variable_manager"].extra_vars.update(extra_vars)
+    script_args = \
+        '''dest_mac=u"{}";dst_ip_addr="{}";ptf_src_port={};ptf_dst_port={};pfc_queue_idx={}'''.format(
+                dest_mac,
+                dst_ip_addr,
+                ptf_src_port,
+                ptf_dst_port,
+                3)
+    supervisor_conf_content = ('''
+[program:{program_name}]
+command=/root/env-python3/bin/ptf --test-dir /root/ptftests/py3 {program_name}.BG_pkt_sender'''
+                               ''' --platform-dir /root/ptftests/ -t'''
+                               ''' '{script_args}' --relax  --platform remote
+process_name={program_name}
+stdout_logfile=/tmp/{program_name}.out.log
+stderr_logfile=/tmp/{program_name}.err.log
+redirect_stderr=false
+autostart=false
+autorestart=true
+startsecs=1
+numprocs=1
+'''.format(script_args=script_args, program_name=program_name))
+    ptfhost.copy(
+        content=supervisor_conf_content,
+        dest=f'/etc/supervisor/conf.d/{program_name}.conf')
+
+    ptfhost.command('supervisorctl reread')
+    ptfhost.command('supervisorctl update')
+    ptfhost.command(f'supervisorctl start {program_name}')
+
+    yield
+
+    try:
+        ptfhost.command(f'supervisorctl stop {program_name}')
+    except BaseException:
+        pass
+    ptfhost.command(f'supervisorctl remove {program_name}')
+
+
```

Comment:
This new fixture looks very similar to the one from #12143. 
Can it be commonized, especially due to its complexity?

If needing autouse is a problem, I'd suggest:
- Either mark the entire class to use the fixture.
- or if lots of classes to mark, make a 'pass' fixture that is marked as "autouse" and simply brings in the common fixture.

Reply:
@rbpittman , I agree with implementing common code, and did check for feasibility. But the new fixture depends on a function defined in the script itself. This issue basically blocked it - unless I move this function also into the common file. I dropped it considering urgency. Dependent function in script: start_background_traffic in timer_accuracy script depends on pfcwd_timer_setup_restore, defined in the same script.

Reply:
I see, how does this sound:
In the other PR:

1. replace the `pfcwd_timer_setup_restore` parameter with "pfc_queue_idx". 
2. Then define the following fixture in that file:
```py
@pytest.fixture
def pfc_queue_idx(pfcwd_timer_setup_restore):
    yield pfcwd_timer_setup_restore['storm_handle'].pfc_queue_idx
```
3. Move the big fixture to common code area
4. Replace `pfcwd_timer_setup_restore['storm_handle'].pfc_queue_idx` in the big fixture with `pfc_queue_idx`
5. In this PR, define the following fixture at the top of the file:
```py
@pytest.fixture
def pfc_queue_idx():
    yield 3   # Hardcoded in the testcase as well.
```

If possible, I'm preferring commonized over delay in getting the PR in since it's quite possible we'll find bugs and have to go fix in 2 locations. Add on double commits and that could cost more time.

Reply:
Implemented.